content/fundamentals/unit-testing.md

```diff
@@ -137,6 +137,8 @@ describe('CatsController', () => {
 });
 ```
 
+> info **Hint** `Test.createTestingModule().compile()` does not call [lifecycle hooks](https://docs.nestjs.com/fundamentals/lifecycle-events) at runtime but they can be called manually for testing purposes.
```

Comment:
Not sure what do you mean here by "can be called manually for testing purposes". If one wants to trigger lifecycle hooks (`onModuleInit` or `onApplicationBootstrap`), the `await app.init()` method must be called. Likewise, to trigger "teardown" lifecycle hooks, the `app.close()` must be used.

Reply:
"Can be called manually" would be doing something like `service.onModuleInit()` inside the test. Useful if you don't want to go through the `init` process, though I don't think it would add too much complexity to do so

Reply:
This may confuse users that if they want to trigger `onModuleInit` on multiple providers, they would have to do it manually for all of them, one by one. The fact that they **can** simply execute the method manually is obvious IMHO so there's no reason to add it to the docs

Reply:
I stumbled upon this one today and I think I agree with @jmcdo29 that it would be nice if the user knows that the testing modules does NOT fire the lifecycle hooks when compiling it and running inside Jest the same way as with regular runtime and all the lifecycle hooks need to be called manually to have the testing modules initialized properly during test.